catalyst_voices/packages/catalyst_voices_services/lib/src/keychain/keychain.dart

```diff
@@ -1,137 +1,17 @@
-import 'package:catalyst_cardano_serialization/catalyst_cardano_serialization.dart';
 import 'package:catalyst_voices_models/catalyst_voices_models.dart';
-import 'package:catalyst_voices_services/src/keychain/key_derivation.dart';
-import 'package:catalyst_voices_services/src/storage/vault/lock_factor.dart';
-import 'package:catalyst_voices_services/src/storage/vault/vault.dart';
-import 'package:logging/logging.dart';
+import 'package:catalyst_voices_services/src/lockable.dart';
+import 'package:flutter/foundation.dart';
 
-const _seedPhraseKey = 'keychain_seed_phrase';
+abstract interface class Keychain implements Lockable {
+  String get id;
 
-/// The keychain which holds the user's [SeedPhrase] and allows
-/// to derive the [Ed25519KeyPair] related to the [AccountRole].
-///
-// TODO(dtscalac): in the future when key derivation algorithm spec
-// will become stable consider to store derived keys instead of deriving
-// them each time they are needed.
+  Future<bool> get isEmpty;
 
-// TODO(damian-molinski): because we have dummy lock factors vault unlocking
-// is dummy too. Any operation on vault require correct lock factor input.
-final class Keychain {
-  final _logger = Logger('Keychain');
+  Future<KeychainMetadata> get metadata;
 
-  final KeyDerivation _keyDerivation;
-  final Vault _vault;
+  Future<Uint8List?> getRootKey();
```

Comment:
Would it be possible to use Ed25519PrivateKey instead of Uin8tList? UInt8List doesn't have any validation for length, checksum, etc., not every Uint8List is a valid Ed25519PrivateKey.

Reply:
Well, you can always call `Ed25519PrivateKey.fromBytes` or `Ed25519PrivateKey.fromHex` but if we're sure it will be class used as root key then yes, we can enforce it

Reply:
In #1041 you have `deriveMasterKey` which returns `Uint8List`. We should aline on naming here. I like `MasterKey` more then `RootKey`

Reply:
In #1041 I don't want to pull catalyst_cardano_serialization into catalyst_key_derivation, most likely we won't communicate directly with catalyst_key_derivation but via a service in catalyst_voices. The services will return a correct type.

Reply:
Master key means the same as root key. Lets use master key then. If you feel like using Ed25519PrivateKey puts too many limits on the code then lets use Uin8tList, it's more versatile and not so strict. But that was the reason I wanted to use a strict type, to avoid passing data which is invalid

Reply:
Used `Ed25519PrivateKey` in `Keychain` 👍